File: app/src/main/java/com/github/goldy1992/mp3player/dagger/modules/service/ExoPlayerModule.java
package com.github.goldy1992.mp3player.dagger.modules.service;

import android.content.Context;

import com.google.android.exoplayer2.C;
import com.google.android.exoplayer2.ExoPlayer;
import com.google.android.exoplayer2.ExoPlayerFactory;
import com.google.android.exoplayer2.SimpleExoPlayer;
import com.google.android.exoplayer2.audio.AudioAttributes;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class ExoPlayerModule {

    @Provides
    @Singleton
    public ExoPlayer provideExoPlayer(Context context) {
        SimpleExoPlayer simpleExoPlayer = ExoPlayerFactory.newSimpleInstance(context);
        AudioAttributes audioAttributes = new AudioAttributes.Builder()
                .setUsage(C.USAGE_MEDIA)
                .setContentType(C.CONTENT_TYPE_MUSIC)
                .build();
        simpleExoPlayer.setAudioAttributes(audioAttributes, true);
        return simpleExoPlayer;
    }
}

File: service/src/main/java/com/github/goldy1992/mp3player/service/dagger/modules/service/ExoPlayerModule.kt
package com.github.goldy1992.mp3player.service.dagger.modules.service

import android.content.Context
import android.media.MediaMetadataRetriever
import android.util.Log
import androidx.media3.common.AudioAttributes
import androidx.media3.common.C
import androidx.media3.common.Player
import androidx.media3.common.util.UnstableApi
import androidx.media3.exoplayer.ExoPlayer
import androidx.media3.exoplayer.RenderersFactory
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.android.components.ServiceComponent
import dagger.hilt.android.qualifiers.ApplicationContext
import dagger.hilt.android.scopes.ServiceScoped
import androidx.annotation.OptIn as AndroidXOptIn

@AndroidXOptIn(UnstableApi::class)
@InstallIn(ServiceComponent::class)
@Module
class ExoPlayerModule  {
    companion object {
        const val LOG_TAG = "ExoPlayerModule"
    }
    @Provides
    @ServiceScoped
    fun provideExoPlayer(@ApplicationContext context: Context,
                         renderersFactory: RenderersFactory
    ): Player {
        Log.v(LOG_TAG, "provideExoPlayer() invoked")
        val audioAttributes = AudioAttributes.Builder()
            .setUsage(C.USAGE_MEDIA)
            .setContentType(C.AUDIO_CONTENT_TYPE_MUSIC)
            .build()
        return ExoPlayer.Builder(context)
            .setRenderersFactory(renderersFactory)
            .setAudioAttributes(audioAttributes, true)
            .build()
    }

    @Provides
    fun providesMediaMetadataRetriever(): MediaMetadataRetriever {
        return MediaMetadataRetriever()
    }


}